TST: io: `test_fortranfiles_mixed_record` fails with numpy 2.0.0

```
________________________ test_fortranfiles_mixed_record ________________________
[gw1] linux -- Python 3.12.1 /opt/hostedtoolcache/Python/3.12.1/x64/bin/python
scipy/io/tests/test_fortran.py:43: in test_fortranfiles_mixed_record
    record = f.read_record('<i4,<f4,<i8,(2)<f8')
        f          = <scipy.io._fortran.FortranFile object at 0x7fdb07867410>
        filename   = '/home/runner/work/scipy/scipy/build-install/lib/python3.12/site-packages/scipy/io/tests/data/fortran-mixed.dat'
scipy/io/_fortran.py:255: in read_record
    dtypes = tuple(np.dtype(dtype) for dtype in dtypes)
        dtype      = None
        dtypes     = ('<i4,<f4,<i8,(2)<f8',)
        first_size = 32
        kwargs     = {}
        self       = <scipy.io._fortran.FortranFile object at 0x7fdb07867410>
scipy/io/_fortran.py:255: in <genexpr>
    dtypes = tuple(np.dtype(dtype) for dtype in dtypes)
        .0         = <tuple_iterator object at 0x7fdb08167910>
        dtype      = '<i4,<f4,<i8,(2)<f8'
/opt/hostedtoolcache/Python/3.12.1/x64/lib/python3.12/site-packages/numpy/_core/_internal.py:197: in _commastring
    warnings.warn(
E   DeprecationWarning: Passing in a parenthesized single number for repeats is deprecated; pass either a single number or indicate a tuple with a comma, like "(2,)".
        astr       = '<i4,<f4,<i8,(2)<f8'
        dtype      = 'f8'
        islist     = True
        mo         = <re.Match object; span=(12, 18), match='(2)<f8'>
        newitem    = 'i8'
        order      = ''
        order1     = ''
        order2     = '<'
        repeats    = '(2)'
        result     = ['i4', 'f4', 'i8']
        startindex = 18
```
https://github.com/scipy/scipy/actions/runs/7296499710/job/19884412452?pr=19731

--- a/scipy/io/tests/test_fortran.py
+++ b/scipy/io/tests/test_fortran.py
@@ -40,7 +40,7 @@ def test_fortranfiles_read():
 def test_fortranfiles_mixed_record():
     filename = path.join(DATA_PATH, "fortran-mixed.dat")
     with FortranFile(filename, 'r', '<u4') as f:
-        record = f.read_record('<i4,<f4,<i8,(2)<f8')
+        record = f.read_record('<i4,<f4,<i8,2<f8')
 
     assert_equal(record['f0'][0], 1)
     assert_allclose(record['f1'][0], 2.3)

